operator/internal/manifests/node_placement_test.go

```diff
@@ -524,3 +525,125 @@ func TestCustomPodAntiAffinity(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomTopologySpreadConstraints(t *testing.T) {
+	template := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"abc": "edf",
+				"gfi": "jkl",
+			},
+			Annotations: map[string]string{
+				"one": "value",
+				"two": "values",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "a-container",
+					Image: "an-image:latest",
+					Env: []corev1.EnvVar{
+						{
+							Name:  "test",
+							Value: "other",
+						},
+					},
+				},
+				{
+					Name:  "b-container",
+					Image: "an-image:latest",
+					Env: []corev1.EnvVar{
+						{
+							Name:  "test",
+							Value: "other",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	spec := &lokiv1.ReplicationSpec{
+		Factor: 2,
+		Zones: []lokiv1.ZoneSpec{
+			{
+				MaxSkew:     1,
+				TopologyKey: "region",
+			},
+			{
+				MaxSkew:     1,
+				TopologyKey: "zone",
```

Comment:
I think I would like to avoid using "region" and "zone" in our examples / tests when using two replication zones in the config. Just to avoid confusion with the correct configuration in public cloud offerings (where region often is already a prefix of zone).

My suggestion would be to pick "datacenter" and "rack" instead to have something "relatable but not real".